### PR TITLE
WIP Adds kourier as a postInstall step for openshift

### PR DESCRIPTION
**Pull request by jmprusi**
Adds kourier as a postinstall step (requires Knative crd's to be created before)

To enable kourier adding the proper ingress.class to the KnativeServing object is enough: 

```yaml
apiVersion: serving.knative.dev/v1alpha1
kind: KnativeServing
metadata:
  name: knative-serving
  namespace: knative-serving
spec:
  config:
   (..)
    network:
      ingress.class: kourier.ingress.networking.knative.dev

```

Currently kourier will always be installed, but we could add an specific trigger to enable/disable it's installation.

### CODE REVIEW

**Comment by mgencur** [comment]
hey @jmprusi , what's the status of this PR please? I went ahead and wanted to check performance of Kourier through this PR but the installation fails for me because the 3scale-kourier-gateway logs an error: 
```
Unable to parse JSON as proto (INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[0].match) safe_regex: Cannot find field.): {"http_filters":[{"name":"envoy.router","config":{}}],"stat_prefix":"stats_server","route_config":{"virtual_hosts":[{"routes":[{"route":{"cluster":"service_stats"},"match":{"safe_regex":{"google_re2":{},"regex":"/(certs|stats(/prometheus)?|server_info|clusters|listeners|ready)?"},"headers":[{"exact_match":"GET","name":":method"}]}}],"domains":["*"],"name":"admin_interface"}]}}
```
Perhaps it's not ready for any testing yet... ?

A few observations so far:
* All Service Mesh pods are installed even if I choose the Kourier ingress. Wouldn't it be better to only install ServiceMesh control plan if I don't choose Kourier (to save a lot of resources)
* It installs 3scale-kourier-control and 3scale-kourier-gateway pods in the knative-serving namespace. I think I like it more than having it in a separate namespace but the fact is that so far we were installing the ingress in a separate one. Just curious what's your thought on that.

File: serving/operator/deploy/resources/kourier/kourier.yaml [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+  namespace: knative-serving
+spec:
+  ports:
+    - name: http2
+      port: 80
+      protocol: TCP
+      targetPort: 8081
+  selector:
+    app: 3scale-kourier-gateway
+  type: ClusterIP
```
Comment (mgencur) [inline serving/operator/deploy/resources/kourier/kourier.yaml]: Upstream has this as LoadBalancer. And all the other services in this file as well. Not sure if this is relevant.. Just making sure you're aware

Comment (davidor) [reply serving/operator/deploy/resources/kourier/kourier.yaml]: We changed this after this PR was opened. We'll need to update this PR with the latest changes.

**Comment by jmprusi** [comment]
@mgencur I've been told to update the PR against master instead of the release-1.3. So I will close this one and open an update one against master. Does it make sense? 

Thanks.

**Comment by mgencur** [comment]
Yes. Makes sense.

**Comment by jmprusi** [comment]
closing in favour of #70

**Pull request closed without merging**